src/skvhttpclient/src/skvhttp/mpack/MPackSerialization.h

```diff
@@ -288,37 +288,54 @@ class MPackNodeReader {
         return true;
     }
 
-    bool _readFromNode(Decimal64& value) {
-        // decimal is packed as a BINARY msgpack type
-        K2LOG_V(log::mpack, "reading decimal64");
+    bool _readFromNode(DecimalD25& value) {
+        // DecimalD25 is memcpy-ed
+        K2LOG_V(log::mpack, "reading decimald25");
         size_t sz;
         const char* data;
 
         if (!_readData(data, sz)) {
             return false;
         }
-        if (sizeof(Decimal64::__decfloat64) != sz) {
+        if (sizeof(DecimalD25) != sz) {
             return false;
         }
 
-        value.__setval(*((Decimal64::__decfloat64*)data));
+        std::memcpy((void *)&value, data, sizeof(DecimalD25));
```

Comment:
I think we can avoid memcpy here - just pass &value to the call to _readData above.